trview.lua.imgui/src/trview.lua.imgui.input.cpp

```diff
@@ -0,0 +1,231 @@
+            {
+                return luaL_error(L, "input_scalarN not yet implemented");
+            }
+
+            int input_text(lua_State* L)
+            {
+                const auto label = get_string(L, 1, "label");
+                std::string out = get_string(L, 1, "value");
+                const auto flags = get_optional_integer(L, 1, "flags");
+                bool result = ImGui::InputText(label.c_str(), &out, flags.value_or(ImGuiInputTextFlags_None));
+                lua_pushboolean(L, result);
+                lua_pushstring(L, out.c_str());
+                return 2;
+            }
+        }
+
+        void register_input(lua_State* L)
+        {
+            // Callbacks are not yet implemented.
+            set_enum(L, "InputTextFlags", -1,
+                {
+                    { "None", ImGuiInputTextFlags_None },
+                    { "CharsDecimal", ImGuiInputTextFlags_CharsDecimal },
+                    { "CharsHexadecimal", ImGuiInputTextFlags_CharsHexadecimal },
+                    { "CharsUppercase", ImGuiInputTextFlags_CharsUppercase },
+                    { "CharsNoBlank", ImGuiInputTextFlags_CharsNoBlank },
+                    { "AutoSelectAll", ImGuiInputTextFlags_AutoSelectAll },
+                    { "EnterReturnsTrue", ImGuiInputTextFlags_EnterReturnsTrue },
+                    { "CallbackCompletion", ImGuiInputTextFlags_CallbackCompletion },
+                    { "CallbackHistory", ImGuiInputTextFlags_CallbackHistory },
+                    { "CallbackAlways", ImGuiInputTextFlags_CallbackAlways },
+                    { "CallbackCharFilter", ImGuiInputTextFlags_CallbackCharFilter },
+                    { "AllowTabInput", ImGuiInputTextFlags_AllowTabInput },
+                    { "CtrlEnterForNewLine", ImGuiInputTextFlags_CtrlEnterForNewLine },
+                    { "NoHorizontalScroll", ImGuiInputTextFlags_NoHorizontalScroll },
+                    { "AlwaysOverwrite", ImGuiInputTextFlags_AlwaysOverwrite },
+                    { "ReadOnly", ImGuiInputTextFlags_ReadOnly },
+                    { "Password", ImGuiInputTextFlags_Password },
+                    { "NoUndoRedo", ImGuiInputTextFlags_NoUndoRedo },
+                    { "CharsScientific", ImGuiInputTextFlags_CharsScientific },
+                    { "CallbackResize", ImGuiInputTextFlags_CallbackResize },
+                    { "CallbackEdit", ImGuiInputTextFlags_CallbackEdit }
+                });
+
+            constexpr luaL_Reg funcs[] = {
+                { "InputDouble", input_double },
+                { "InputFloat", input_float },
+                { "InputFloat2", input_float2 },
+                { "InputFloat3", input_float3 },
+                { "InputFloat4", input_float4 },
+                { "InputInt", input_int },
+                { "InputInt2", input_int2 },
+                { "InputInt3", input_int3 },
+                { "InputInt4", input_int4 },
+                // { "InputScalar", input_scalar },
```

Comment:
Just checking if you want to leave these in, I noticed other commented lines had been removed in this file.

Reply:
Thought I'd leave them here as I should probably add them at some point, I just don't know how to do it at the moment.